src/nvim/ui_client.c

```diff
@@ -153,7 +153,6 @@ void ui_client_event_grid_line(Array args)
   Integer endcol, clearcol;
   // TODO(hlpr98): Accomodate other LineFlags when included in grid_line
   LineFlags lineflags = 0;
-  endcol = startcol;
 
```

Comment:
I would go further and move the entire declaration of `endcol` down to where it's first initialized at line 204.